systemtest/infra_pubsublite.go

```diff
@@ -0,0 +1,94 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package systemtest
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+
+	"github.com/hashicorp/terraform-exec/tfexec"
+)
+
+var (
+	googleProject string
+	googleRegion  string
+)
+
+// PubSubLiteConfig for PubSubLite topics, subscriptions, reservations
+type PubSubLiteConfig struct {
+	// TFPath is the path where the terraform files are present.
+	TFPath string
+	// Project is the GCP Project name.
+	Project string
+	// Region is the GCP region.
+	Region string
+	// Topics (and subscriptions) to create.
+	Topics []string
+	// ReservationSuffix to use.
+	ReservationSuffix string
+}
+
+func newPubSubLiteConfig(topics ...string) PubSubLiteConfig {
+	return PubSubLiteConfig{
+		Topics:  topics,
+		Project: googleProject,
+		Region:  googleRegion,
+		TFPath:  filepath.Join("tf", "pubsublite"),
+	}
+}
+
+// ProvisionPubSubLite provisions the PubSubLite resources. The specified
+// terraform path must accept at least these variables:
+// - project
+// - region
+// - suffix (used to suffix the reservation)
+// - topics
+func ProvisionPubSubLite(ctx context.Context, cfg PubSubLiteConfig) error {
+	if cfg.ReservationSuffix == "" {
+		cfg.ReservationSuffix = persistentSuffix
+	}
+	logger().Infof("provisioning PubSubLite infrastructure with config: %+v", cfg)
+	tf, err := NewTerraform(ctx, cfg.TFPath)
+	if err != nil {
+		return err
```

Comment:
suggestion: should we wrap this and other errors in the provisioning code ?

Reply:
The errors were easy to enough to trace as it is right now, given the logging. I'd rather not unless it becomes a problem.